worker/src/d1/mod.rs

```diff
@@ -150,6 +150,32 @@ impl D1PreparedStatement {
         }
     }
 
+    /// Bind one or more parameters to the statement.
+    /// Returns a new statement with the bound parameters, leaving the old statement available for reuse.
+    pub fn bind_refs(&self, values: &[&JsValue]) -> Result<Self> {
+        let array: Array = values.iter().collect::<Array>();
+
+        match self.0.bind(array) {
+            Ok(stmt) => Ok(D1PreparedStatement(stmt)),
+            Err(err) => Err(Error::from(err)),
+        }
+    }
+
+    /// Bind a batch of parameter values, returning a batch of prepared statements.
+    /// Result can be passed to [`D1Database::batch`] to execute the statements.
+    pub fn batch_bind(&self, values: &[&[&JsValue]]) -> Result<Vec<Self>> {
+        values
+            .iter()
+            .map(|batch| {
+                let array: Array = batch.iter().collect::<Array>();
+                match self.0.bind(array) {
+                    Ok(stmt) => Ok(D1PreparedStatement(stmt)),
+                    Err(err) => Err(Error::from(err)),
+                }
+            })
+            .collect()
+    }
```

Comment:
To match my tweaks to bind_refs, and make it a little more DRY, I would change this to
```rs
pub fn batch_bind<'a, T: 'a, U: 'a>(&self, values: T) -> Result<Vec<Self>>
where T : IntoIterator<Item = &'a &'a U>,
  &'a U : IntoIterator<Item = &'a &'a JsValue> {
    values
        .into_iter()
        .map(|&batch| {
            self.bind_refs(batch)
        })
        .collect()
}
```
Rust is clever about mapping the Vec<Result<Self>> to a Result<Vec<Self>> so we might as well use it